configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionSerializer.java

```diff
@@ -53,20 +59,30 @@ void write(SessionZooKeeperClient zooKeeperClient, ApplicationId applicationId,
     }
 
     SessionData read(SessionZooKeeperClient zooKeeperClient, BooleanFlag readSessionData) {
-        if (readSessionData.value())
-            return zooKeeperClient.readSessionData();
-        else
-            return new SessionData(zooKeeperClient.readApplicationId(),
-                                   zooKeeperClient.readApplicationPackageReference(),
-                                   zooKeeperClient.readVespaVersion(),
-                                   zooKeeperClient.readCreateTime(),
-                                   zooKeeperClient.readDockerImageRepository(),
-                                   zooKeeperClient.readAthenzDomain(),
-                                   zooKeeperClient.readQuota(),
-                                   zooKeeperClient.readTenantSecretStores(),
-                                   zooKeeperClient.readOperatorCertificates(),
-                                   zooKeeperClient.readCloudAccount(),
-                                   zooKeeperClient.readDataplaneTokens());
+        if (readSessionData.value() && zooKeeperClient.sessionDataExists())
+            try {
+                return zooKeeperClient.readSessionData();
+            } catch (Exception e) {
+                log.log(WARNING, "Unable to read session dato for session " + zooKeeperClient.sessionId() +
+                        ": " + Exceptions.toMessageString(e));
+                readSessionDataFromLegacyPaths(zooKeeperClient);
```

Comment:
Unnecessary read?

Reply:
Yeah, I guess I can just let if fall through, will remove it